### PR TITLE
add loader

### PR DESCRIPTION
**Pull request by PPWangyc**
1. Add.a pytorch Dataloader
2. Padding the spikes_sparse_data

Sample file in /src/main.py

### CODE REVIEW

**Comment by colehurwitz** [comment]
Looks great! Small comments:
- ritght --> right
- Are we happy with BaseDataset as the name? Should we call this a SpikingDataset? Or should we leave it general so we can add more modalities in the future?

**Comment by colehurwitz** [comment]
Nice changes! Quick questions about max_length:
1. The binning we do for the spiking data gives us about 50 - 100 time bins (right @yzhang511 ?) so 5000 max_length seems very aggressive. Not sure we want to force everything to be so long. Maybe max_length=100.
2. Should the padding be dynamic? In other words, should we pad to max length of the batch rather than predefine this?

**Comment by yzhang511** [comment]
Yes, I am currently using a time interval of 2 seconds and bin size of 20
ms so 100 time bins.

**Comment by PPWangyc** [comment]
> Nice changes! Quick questions about max_length:
> 
> 1. The binning we do for the spiking data gives us about 50 - 100 time bins (right @yzhang511 ?) so 5000 max_length seems very aggressive. Not sure we want to force everything to be so long. Maybe max_length=100.
> 2. Should the padding be dynamic? In other words, should we pad to max length of the batch rather than predefine this?

1. Yes, we should change this later. Let's discuss the hyperparameters in the meeting, and I'll add the argparser/config as well. 
2. People always make it fixed(I think). It does not affect the training performance a lot.

**Comment by colehurwitz** [comment]
Merging!
